filters/auth/jwt_validation.go

```diff
@@ -0,0 +1,170 @@
+package auth
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/MicahParks/keyfunc"
+	"github.com/golang-jwt/jwt"
+	"github.com/opentracing/opentracing-go"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	JwtValidationName = "jwtValidation"
+)
+
+type (
+	jwtValidationSpec struct {
+		options TokenintrospectionOptions
+	}
+
+	jwtValidationFilter struct {
+		jwksUri string
+	}
+)
+
+var jwksMap map[string]*keyfunc.JWKs = make(map[string]*keyfunc.JWKs)
```

Comment:
Please add a doc string describing what is the key in this map.

Also the map is global var and access is not guarded with a mutex, I guess this follows e.g.
https://github.com/zalando/skipper/blob/2059778e1924aaafb86accfc0b27e10245f13117/filters/auth/tokenintrospection.go#L72

Lets leave it as is for now, I would keep it scoped within the `jwtValidationSpec` and guard access with a mutex.